homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,187 @@
+}
+
+DEVICE_CLASSES = {
+    CONF_CELSIUS: DEVICE_CLASS_PRESSURE,
+    CONF_KELVIN: DEVICE_CLASS_PRESSURE,
+    CONF_BAR: DEVICE_CLASS_PRESSURE,
+    CONF_SECONDS: DEVICE_CLASS_TIMESTAMP,
+    CONF_HOURS: DEVICE_CLASS_TIMESTAMP,
+    CONF_TIMESTAMP: DEVICE_CLASS_TIMESTAMP,
+}
+
+UNITS = {
+    CONF_CELSIUS: TEMP_CELSIUS,
+    CONF_SECONDS: TIME_SECONDS,
+    CONF_KELVIN: TEMP_KELVIN,
+    CONF_BAR: PRESSURE_BAR,
+    CONF_PERCENT: PERCENTAGE_PERCENT,
+    CONF_ENERGY: ENERGY_KILO_WATT_HOUR,
+    CONF_VOLTAGE: VOLTAGE_VOLT,
+    CONF_HOURS: TIME_HOUR,
+    CONF_FLOW: FLOW_LITERS_PER_MINUTE,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DEVICE_CLASS = "sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_SENSORS): vol.All(
+            cv.ensure_list,
+            [
+                {
+                    vol.Required(CONF_GROUP): vol.All(
+                        cv.string,
+                        vol.Any(CONF_PARAMETERS, CONF_CALCULATIONS, CONF_VISIBILITIES),
+                    ),
+                    vol.Required(CONF_ID): cv.string,
+                    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+                    vol.Optional(CONF_ICON): cv.string,
+                }
+            ],
+        )
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Luxtronik sensor."""
+    luxtronik = hass.data.get(DATA_LUXTRONIK)
+    if not luxtronik:
+        return False
+
+    sensors = config.get(CONF_SENSORS)
+
+    entities = []
+    for sensor_cfg in sensors:
+        sensor = luxtronik.get_sensor(sensor_cfg[CONF_GROUP], sensor_cfg[CONF_ID])
+        if sensor:
+            entities.append(LuxtronikSensor(luxtronik, sensor, sensor_cfg))
+        else:
+            _LOGGER.warning(
+                "Invalid Luxtronik ID %s in group %s",
+                sensor_cfg[CONF_ID],
+                sensor_cfg[CONF_GROUP],
+            )
+
+    add_entities(entities, True)
+
+
+class LuxtronikSensor(Entity):
+    """Representation of a Luxtronik sensor."""
+
+    def __init__(self, luxtronik, sensor, sensor_cfg):
+        """Initialize a new Luxtronik sensor."""
+        self._luxtronik = luxtronik
+        self._sensor = sensor
+        self._name = sensor_cfg[CONF_FRIENDLY_NAME]
+        self._icon = sensor_cfg[CONF_ICON]
+
+    @property
+    def entity_id(self):
+        """Return the entity_id of the sensor."""
+        if not self._name:
+            return ENTITY_ID_FORMAT.format(slugify(self._sensor.name))
+        return ENTITY_ID_FORMAT.format(slugify(self._name))
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if not self._name:
+            return self._sensor.name
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        if not self._icon:
+            return ICONS.get(self._sensor.measurement_type, None)
```

Comment:
`dict.get()` returns None by default so you do not need to specify it. Please remove `None` here and in the all other cases.